internal/acctest/config/oauth/accesstokenmapping/oauth_access_token_mapping_test.go

```diff
@@ -0,0 +1,352 @@
+	resourceName := "myOauthAccessTokenMapping"
+
+	// Initial Model
+	// attributeContractFulfillment
+	initialAttributeContractFulfillmentVal := map[string]client.AttributeFulfillmentValue{}
+	initialAFV := client.NewAttributeFulfillmentValue(*client.NewSourceTypeIdKey("TEXT"), "Administrator")
+	initialAttributeContractFulfillmentVal["extended_contract"] = *initialAFV
+
+	initialResourceModel := oauthAccessTokenMappingResourceModel{
+		attributeContractFulfillment: initialAttributeContractFulfillmentVal,
+	}
+
+	//  Updated Model
+	// attributeContractFulfillment
+	updatedAttributeContractFullfillmentVal := map[string]client.AttributeFulfillmentValue{}
+	updatedAFV := client.NewAttributeFulfillmentValue(*client.NewSourceTypeIdKey("TEXT"), "Administrator2")
+	updatedAttributeContractFullfillmentVal["extended_contract"] = *updatedAFV
+
+	//  attributeSource
+	jdbcAttributeSource := client.NewJdbcAttributeSource("ADMINISTRABLE_ROLE_AUTHORIZATIONS", "${client_id}", "JDBC", *client.NewResourceLink("ProvisionerDS"))
+	jdbcAttributeSource.Schema = pointers.String("INFORMATION_SCHEMA")
+	jdbcAttributeSource.ColumnNames = []string{"GRANTEE", "IS_GRANTABLE", "ROLE_NAME"}
+	jdbcAttributeSource.Id = pointers.String("test")
+	jdbcAttributeSource.Description = pointers.String("test")
+
+	issuanceCriteria := client.NewIssuanceCriteria()
+	conditionalIssuanceCriteriaEntry := client.NewConditionalIssuanceCriteriaEntry(*client.NewSourceTypeIdKey("CONTEXT"), "ClientId", "EQUALS_CASE_INSENSITIVE", "text")
+	issuanceCriteria.ConditionalCriteria = append(issuanceCriteria.ConditionalCriteria, *conditionalIssuanceCriteriaEntry)
+
+	updatedResourceModel := oauthAccessTokenMappingResourceModel{
+		attributeContractFulfillment: updatedAttributeContractFullfillmentVal,
+		attributeSource:              jdbcAttributeSource,
+		issuanceCriteria:             issuanceCriteria,
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.ConfigurationPreCheck(t) },
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"pingfederate": providerserver.NewProtocol6WithError(provider.NewTestProvider()),
+		},
+		CheckDestroy: testAccCheckOauthAccessTokenMappingDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOauthAccessTokenMapping(resourceName, initialResourceModel),
+				Check:  testAccCheckExpectedOauthAccessTokenMappingAttributes(initialResourceModel),
+			},
+			{
+				// Test updating some fields
+				Config: testAccOauthAccessTokenMapping(resourceName, updatedResourceModel),
+				Check:  testAccCheckExpectedOauthAccessTokenMappingAttributes(updatedResourceModel),
+			},
+			{
+				// Test importing the resource
+				Config:            testAccOauthAccessTokenMapping(resourceName, updatedResourceModel),
+				ResourceName:      "pingfederate_oauth_access_token_mapping." + resourceName,
+				ImportStateId:     oauthAccessTokenMappingId,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
```

Comment:
Should we have a delete drift check in here?

Reply:
whoops! got lost in re-work